internal/transactionsubmission/transaction_worker.go

```diff
@@ -184,54 +192,43 @@ func (tw *TransactionWorker) runJob(ctx context.Context, txJob *TxJob) error {
 func (tw *TransactionWorker) handleFailedTransaction(ctx context.Context, txJob *TxJob, hTxResp horizon.Transaction, hErr *utils.HorizonErrorWrapper) error {
 	log.Ctx(ctx).Errorf("🔴 Error processing job: %v", hErr)
 
-	metricsMetadata := tssMonitor.TxMetadata{
-		EventID:          tw.jobUUID,
-		SrcChannelAcc:    txJob.ChannelAccount.PublicKey,
-		PaymentEventType: sdpMonitor.PaymentMarkedForReprocessingLabel,
+	handler, err := tw.handlerFactory.GetHandler(&txJob.Transaction)
+	if err != nil {
+		return fmt.Errorf("getting transaction handler: %w", err)
 	}
-	defer func() {
-		tw.monitorSvc.LogAndMonitorTransaction(
-			ctx,
-			txJob.Transaction,
-			sdpMonitor.PaymentErrorTag,
-			metricsMetadata,
-		)
-	}()
 
-	err := tw.saveResponseXDRIfPresent(ctx, txJob, hTxResp)
+	isRetryable := !(hErr.IsHorizonError() && hErr.ShouldMarkAsError())
+	defer handler.MonitorTransactionProcessingFailed(ctx, txJob, tw.jobUUID, isRetryable, hErr.Error())
+
+	err = tw.saveResponseXDRIfPresent(ctx, txJob, hTxResp)
 	if err != nil {
 		return fmt.Errorf("saving response XDR: %w", err)
 	}
 
 	var hErrWrapper *utils.HorizonErrorWrapper
 	if errors.As(hErr, &hErrWrapper) {
```

Comment:
By reviewing your code I realized `if errors.As(hErr, &hErrWrapper) {` is not required at all, since hErr is already of the desired type. I believe this is a leftover from a refactor.

Do you mind removing the `if` statement, as well as the redundant `hErrWrapper` variable?